File: go/wam/8248.go
const WamWaffleCompanionStateLifecycle = 8248 // channel: regular

type WaffleCompanionStateLifecycleEvent struct {
	WaffleLifecycleElapsedMs int64 `wam:"waffle_lifecycle_elapsed_ms"` // field 1
	WaffleLifecycleErrorAction WAFFLELIFECYCLEERRORACTIONTYPE `wam:"waffle_lifecycle_error_action"` // field 2
	WaffleLifecycleErrorCode WAFFLELIFECYCLEERRORCODETYPE `wam:"waffle_lifecycle_error_code"` // field 3
	WaffleLifecycleHasAccessToken bool `wam:"waffle_lifecycle_has_access_token"` // field 4
	WaffleLifecycleHasExistingRow bool `wam:"waffle_lifecycle_has_existing_row"` // field 5
	WaffleLifecycleLinkState WAFFLELIFECYCLELINKSTATETYPE `wam:"waffle_lifecycle_link_state"` // field 6
	WaffleLifecycleRetryCount int64 `wam:"waffle_lifecycle_retry_count"` // field 7
	WaffleLifecycleTraceAction WAFFLELIFECYCLETRACEACTIONTYPE `wam:"waffle_lifecycle_trace_action"` // field 8
	WaffleLifecycleTraceSource WAFFLELIFECYCLETRACESOURCETYPE `wam:"waffle_lifecycle_trace_source"` // field 9
	WaffleLifecycleUnlinkType int64 `wam:"waffle_lifecycle_unlink_type"` // field 10
}

File: rust/wam/8248.rs
pub const WAM_WAFFLE_COMPANION_STATE_LIFECYCLE: u32 = 8248;

#[derive(Debug, Default)]
pub struct WaffleCompanionStateLifecycleEvent {
    /// field 1, wire `waffle_lifecycle_elapsed_ms`
    pub waffle_lifecycle_elapsed_ms: Option<i64>,
    /// field 2, wire `waffle_lifecycle_error_action`
    pub waffle_lifecycle_error_action: Option<WAFFLELIFECYCLEERRORACTIONTYPE>,
    /// field 3, wire `waffle_lifecycle_error_code`
    pub waffle_lifecycle_error_code: Option<WAFFLELIFECYCLEERRORCODETYPE>,
    /// field 4, wire `waffle_lifecycle_has_access_token`
    pub waffle_lifecycle_has_access_token: Option<bool>,
    /// field 5, wire `waffle_lifecycle_has_existing_row`
    pub waffle_lifecycle_has_existing_row: Option<bool>,
    /// field 6, wire `waffle_lifecycle_link_state`
    pub waffle_lifecycle_link_state: Option<WAFFLELIFECYCLELINKSTATETYPE>,
    /// field 7, wire `waffle_lifecycle_retry_count`
    pub waffle_lifecycle_retry_count: Option<i64>,
    /// field 8, wire `waffle_lifecycle_trace_action`
    pub waffle_lifecycle_trace_action: Option<WAFFLELIFECYCLETRACEACTIONTYPE>,
    /// field 9, wire `waffle_lifecycle_trace_source`
    pub waffle_lifecycle_trace_source: Option<WAFFLELIFECYCLETRACESOURCETYPE>,
    /// field 10, wire `waffle_lifecycle_unlink_type`
    pub waffle_lifecycle_unlink_type: Option<i64>,
}